torch/_dynamo/debug_utils.py

```diff
@@ -962,33 +966,51 @@ def dynamo_accuracy_minifier_backend(gm, example_inputs, compiler_name):
 
     from torch._dynamo.optimizations.backends import BACKENDS
 
+    # (1) Get the backend.
     if compiler_name == "inductor":
         from torch._inductor.compile_fx import compile_fx
 
         compiler_fn = compile_fx
     else:
+        assert (
+            compiler_name in BACKENDS
+        ), f"Unknown compiler name {compiler_name} provided."
         compiler_fn = BACKENDS[compiler_name]
 
-    # Set the eval mode to remove randomness.
-    gm.eval()
+    def backend_accuracy_fwd(*args):
+        # (2) Pass it to accuracy eval, we compile within this func.
+        if backend_accuracy_fails(
+            gm, example_inputs=example_inputs, real_args=args, compiler_fn=compiler_fn
+        ):
+            # (3) Accuracy failed - this is good, we are trying to repro this.
+            log.warning("Accuracy failed for the TorchDyanmo produced graph")
+            dump_state_fn = functools.partial(
+                dump_backend_state, compiler_name=compiler_name, check_accuracy=True
+            )
 
-    # Check Accuracy
-    if backend_accuracy_fails(gm, example_inputs, compiler_fn):
-        log.warning("Accuracy failed for the TorchDyanmo produced graph")
-        dump_state_fn = functools.partial(
-            dump_backend_state, compiler_name=compiler_name, check_accuracy=True
-        )
-        fails_fn = functools.partial(
-            backend_accuracy_fails,
-            compiler_fn=compiler_fn,
-        )
-        dump_state_fn(fx.GraphModule(gm, copy.deepcopy(gm.graph)), example_inputs)
-        minifier(
-            gm,
-            example_inputs,
-            module_fails=fails_fn,
-            dump_state=dump_state_fn,
-        )
-    else:
-        log.error("Input graph does not fail accuracy testing")
-    return gm
+            # (4) Bind the current compiler_fn and real_args to accuracy_eval
+            # We need to do this in order to ensure that (a) we correctly recompile on every
+            # minification, and (b) that the args captured here, in the runtime, are passed to eval
```

Comment:
This comment is backwards. To "correctly recompile on every minification", the bug in https://github.com/pytorch/pytorch/pull/90089/files/ae9f856428538468b5da8825e9d796790b010db6#r1038871303 was that you didn't pass `backend_accuracy_fails` to the minifier at all. Binding compiler_fn/real_args has nothing to do with ensuring that we actually recompile; you must bind them for the simpler reason that the minifier isn't going to pass those arguments directly lol

Reply:
> This comment is backwards. To "correctly recompile on every minification", the bug in https://github.com/pytorch/pytorch/pull/90089/files/ae9f856428538468b5da8825e9d796790b010db6#r1038871303 was that you didn't pass `backend_accuracy_fails` to the minifier at all. Binding compiler_fn/real_args has nothing to do with ensuring that we actually recompile; you must bind them for the simpler reason that the minifier isn't going to pass those arguments directly lol

I was passing a different function, identical to this except for recompilation. So now I'm confused by what you mean.

Reply:
The comment says "Bind the current compiler_fn and real_args to accuracy_eval; we need to do this". I parse this as "we need to [Bind the current compiler_fn and real_args]". Let's suppose I removed this binding. Then I would expect this code to fail with "not enough parameters passed".